Index: src/main/java/com/jagrosh/jmusicbot/audio/AudioHandler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/jagrosh/jmusicbot/audio/AudioHandler.java b/src/main/java/com/jagrosh/jmusicbot/audio/AudioHandler.java
--- a/src/main/java/com/jagrosh/jmusicbot/audio/AudioHandler.java	(revision bc64b45d5c290705a64972dde6da4ae040d21e10)
+++ b/src/main/java/com/jagrosh/jmusicbot/audio/AudioHandler.java	(date 1624634451799)
@@ -15,6 +15,7 @@
  */
 package com.jagrosh.jmusicbot.audio;
 
+import com.jagrosh.jmusicbot.Bot;
 import com.jagrosh.jmusicbot.JMusicBot;
 import com.jagrosh.jmusicbot.playlist.PlaylistLoader.Playlist;
 import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
@@ -22,15 +23,28 @@
 import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
 import com.sedmelluq.discord.lavaplayer.track.AudioTrackEndReason;
 import com.sedmelluq.discord.lavaplayer.track.playback.AudioFrame;
-import java.util.HashSet;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Set;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+
+import net.dv8tion.jda.api.entities.Activity;
+import org.json.JSONException;
+import org.json.JSONObject;
+
 import com.jagrosh.jmusicbot.queue.FairQueue;
 import com.jagrosh.jmusicbot.settings.Settings;
 import com.jagrosh.jmusicbot.utils.FormatUtil;
 import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack;
 import java.nio.ByteBuffer;
+import java.util.regex.Pattern;
+
 import net.dv8tion.jda.api.EmbedBuilder;
 import net.dv8tion.jda.api.JDA;
 import net.dv8tion.jda.api.MessageBuilder;
@@ -52,7 +66,8 @@
     private final PlayerManager manager;
     private final AudioPlayer audioPlayer;
     private final long guildId;
-    
+    private Timer timer;
+
     private AudioFrame lastFrame;
 
     protected AudioHandler(PlayerManager manager, Guild guild, AudioPlayer player)
@@ -164,6 +179,10 @@
         {
             if(!playFromDefault())
             {
+                if(timer != null){
+                    this.timer.cancel();
+                    this.timer = null;
+                }
                 manager.getBot().getNowplayingHandler().onTrackUpdate(guildId, null, this);
                 if(!manager.getBot().getConfig().getStay())
                     manager.getBot().closeAudioConnection(guildId);
@@ -183,7 +202,17 @@
     public void onTrackStart(AudioPlayer player, AudioTrack track) 
     {
         votes.clear();
-        manager.getBot().getNowplayingHandler().onTrackUpdate(guildId, track, this);
+        if(track.getInfo().uri.contains("anison") && this.timer == null){
+            try {
+                this.timer = new Timer(true);
+                timer.schedule(new AnisonUpdateTask(this.manager.getBot()), 0, 10000);
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            }
+        }
+        else{
+            manager.getBot().getNowplayingHandler(). onTrackUpdate(guildId, track, this);
+        }
     }
 
     
@@ -209,7 +238,15 @@
 
             try 
             {
-                eb.setTitle(track.getInfo().title, track.getInfo().uri);
+                String title;
+                if(track.getInfo().uri.contains("anison.fm")){
+                    title = jda.getPresence().getActivity().getName();
+                }
+                else
+                {
+                    title = track.getInfo().title;
+                }
+                eb.setTitle(title, track.getInfo().uri);
             }
             catch(Exception e) 
             {
@@ -311,4 +348,42 @@
     {
         return jda.getGuildById(guildId);
     }
+
+    private class AnisonUpdateTask extends TimerTask{
+        private final Bot bot;
+        private final URL url;
+        private final Pattern REMOVE_TAGS = Pattern.compile("<.+?>");
+        public AnisonUpdateTask(Bot bot) throws MalformedURLException {
+            this.bot = bot;
+            this.url = new URL("https://anison.fm/status.php?widget=true");
+        }
+
+        @Override
+        public void run() {
+            try {
+                JSONObject json = this.readJsonFromUrl(this.url);
+                String on_air = REMOVE_TAGS.matcher(json.getString("on_air")).replaceAll("");
+                String s = on_air.replaceAll("В эфире: ", "").replaceAll("&#151;", "—");
+                bot.getJDA().getPresence().setActivity(Activity.listening(s));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        private String readAll(Reader rd) throws IOException {
+            StringBuilder sb = new StringBuilder();
+            int cp;
+            while ((cp = rd.read()) != -1) {
+                sb.append((char) cp);
+            }
+            return sb.toString();
+        }
+        public JSONObject readJsonFromUrl(URL url) throws IOException, JSONException {
+            try (InputStream is = url.openStream()) {
+                BufferedReader rd = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
+                String jsonText = readAll(rd);
+                return new JSONObject(jsonText);
+            }
+        }
+    }
 }
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision bc64b45d5c290705a64972dde6da4ae040d21e10)
+++ b/pom.xml	(date 1624632215713)
@@ -62,6 +62,11 @@
             <version>1.3.2</version>
         </dependency>
         <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>20180813</version>
+        </dependency>
+        <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <version>4.13.1</version>
